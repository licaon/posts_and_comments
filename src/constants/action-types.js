export const INIT = 'INIT';

export const START_LOADING_DATA = 'START_LOADING_DATA';
export const END_LOADING_DATA = 'END_LOADING_DATA';

export const GET_POSTS = 'GET_POSTS';
export const ADD_POST = 'ADD_POST';
export const DELETE_POST = 'DELETE_POST';

export const ADD_SEARCH_FILTER = 'ADD_SEARCH_FILTER';
export const ADD_SELECT_FILTER = 'ADD_SELECT_FILTER';

export const GET_USERS = 'GET_USERS';
export const ADD_USER = 'ADD_USER';
