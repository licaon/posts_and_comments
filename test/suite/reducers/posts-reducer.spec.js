import {  ADD_POST, DELETE_POST } from 'src/constants/action-types';
import PostsReducer from 'src/reducers/posts-reducer';

describe('PostsReducer', () => {
  it(`should update reducer correctly on ${ADD_POST}`, (done) => {
    expect(PostsReducer(
      {
        posts: [{
          id: 1,
        }],
      },
      {
        type: ADD_POST,
        payload: {
          id: 2,
        },
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should update reducer correctly on ${DELETE_POST}`, (done) => {
    expect(PostsReducer(
      {
        posts: [
          {
            id: 1,
          },
          {
            id: 2,
          },
        ],
      },
      {
        type: DELETE_POST,
        payload: {
          id: 1,
        },
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should return reducer correctly on unknown action`, (done) => {
    expect(PostsReducer(
      undefined,
      { type: 'unknown' },
    )).toMatchSnapshot();
    done();
  });
});
