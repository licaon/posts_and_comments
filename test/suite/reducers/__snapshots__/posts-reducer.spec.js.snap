// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsReducer should return reducer correctly on unknown action 1`] = `
Object {
  "posts": Array [],
}
`;

exports[`PostsReducer should update reducer correctly on ADD_POST 1`] = `
Object {
  "posts": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
}
`;

exports[`PostsReducer should update reducer correctly on DELETE_POST 1`] = `
Object {
  "posts": Array [
    Object {
      "id": 2,
    },
  ],
}
`;
